x/feemarket/keeper/abci.go

```diff
@@ -58,25 +59,35 @@ func (k *Keeper) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) {
 		return
 	}
 
-	gasWanted := k.GetTransientGasWanted(ctx)
-	gasUsed := ctx.BlockGasMeter().GasConsumedToLimit()
+	gasWanted := sdkmath.NewIntFromUint64(k.GetTransientGasWanted(ctx))
+	gasUsed := sdkmath.NewIntFromUint64(ctx.BlockGasMeter().GasConsumedToLimit())
+
+	if !gasWanted.IsInt64() {
+		k.Logger(ctx).Error(fmt.Sprintf("integer overflow by integer type conversion. gasWanted %s is higher than MaxInt64", gasWanted))
```

Comment:
nit, for logs you usually use `(msg, key, value)`. Example:

```suggestion
		k.Logger(ctx).Error("integer overflow by integer type conversion. Gas wanted > MaxInt64",  "gas wanted", gasWanted.String()))
```